### PR TITLE
Feature/player stats initial

### PR DESCRIPTION
**Pull request by bar0net**
- Player attributes extracted to a struct (PlayerStats)
- Player gets 2 methods, Equip and UnEquip that pass a PlayerStats parameter to the player. The players' attributes are updated according to the specified parameters.
- Player Stats are exposed to the editor and are serializable
- Added health and mana regen per second. 
- Added an "out of combat" timer to trigger health regen
- Added a Player Stats Debug section to the editor for debug purposes.

Test:
- F1/F3 equip an item that increases max hp/mana by 10 
  - Max Hp/Mana should increase by 10
  - Current Hp/Mana should not increase (by the effect of the item)

- F2/F4 unequip an item that increases max hp/mana by 10 
  - Max Hp/Mana should decrease by 10
  - current Hp/Mana should be clipped by max Hp/Mana

- If current hp/mana are less than max, it should regen at a default rate of 5 pts/sec
- Stats serialize correctly when saving/loading a scene (max health, max mana, dexterity, strength, hp regen and mana regen, current health and current mana are out of scope)

- The player should regenerate hp if it's not full and has not received damage in the last 3 seconds (default)
- The player should regenerate mana if not full and not used mana recently

### CODE REVIEW

**Comment by AlfonsoMestres** [comment]
As a Designer, I want Health and Mana to slowly regenerate. 
- Regenerate when not in combat for 10 seconds*****.
- Regeneration should be slower to promote the use of consumables. (Like 3 points / second or so*****)

*** Those values are the first approach and they might be changed later.**

**Comment by bar0net** [comment]
You should be able to define the "out of combat time", Hp regeneration (per second) and mana regeneration (per second) in the editor.

File: Scripts/PlayerMovement/PlayerMovement.h [hunk 1]
```diff
@@ -45,6 +45,44 @@ class PlayerStateUppercut;
...
+	}
+
+public:
+	float health = 0;
+	float mana = 0;
+	int strength  = 0;
+	int dexterity = 0;
+
+	float hpRegen = 0;
+	float manaRegen = 0;
```
Comment (AlfonsoMestres) [inline Scripts/PlayerMovement/PlayerMovement.h]: Codestyling -> 0.0f

Comment (bar0net) [reply Scripts/PlayerMovement/PlayerMovement.h]: Corrections made as instructed

**Pull request merged**